build.zig

```diff
@@ -28,20 +30,24 @@ pub fn build(b: *Builder) !void {
 
     const fmt_step = b.addFmt(&[_][]const u8{
         "build.zig",
+        "mkfat32.zig",
         "src",
         "test",
     });
     b.default_step.dependOn(&fmt_step.step);
 
     const main_src = "src/kernel/kmain.zig";
     const arch_root = "src/kernel/arch";
+    const fat32_bootstrap_src = "fat32_bootstrap.asm";
```

Comment:
What's the purpose of the fat32_bootstrap?

Reply:
It is part of the FAT32 header. Is it used if it is loaded as a bootloader and will just print "please insert a bootable media, press any key to restart." It is taken from my bootloader. I guess it isn't 100% needed, but it part of the specification so may not be able to mount in Linux. There could just be a random flat binary, but the code must come from somewhere.

Reply:
That makes sense. I think it would be nice to move it somewhere under arch/x86.

Reply:
It shouldn't go into the x86 folder as it can be used for any arch. Maybe could add the pre-build binary to the git repo and remove the build step for this, so removing the nasm dependency. And if people want to build it, they can separetly

Reply:
I think that makes a lot of sense 👍 Building a FAT image shouldn't be a dependency for the kernel.